python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -674,19 +674,54 @@ def do_evaluate(
     ) -> Column:
         """Evaluate this expression given a dataframe for context."""
         if self.name == pl_expr.StringFunction.Contains:
-            child, pattern = self.children
+            child, arg = self.children
             column = child.evaluate(df, context=context, mapping=mapping)
-            assert isinstance(pattern, Literal)
 
             literal, _ = self.options
             if literal:
-                return Column(plc.strings.find.contains(column.obj, pattern.value))
+                val = (
+                    arg.value
+                    if isinstance(arg, Literal)
+                    else arg.evaluate(df, context=context, mapping=mapping).obj
+                )
+                return Column(plc.strings.find.contains(column.obj, val))
```

Comment:
This bit doesn't work. `arg.value` is a pyarrow object for a `Literal` which can't be passed to contains

Try:

```
pat = arg.evaluate(df, context=context, mapping=mapping)
pattern = pat.obj_scalar if pat.is_scalar and pat.obj.size() != column.obj.size() else pat.obj
return Column(plc.strings.find.contains(column.obj, pattern))
```
I think.